HACKING.md

```diff
@@ -318,65 +318,152 @@ the build frontend is [build](https://pypi.org/project/build/).
-3. The "Release Title" is the full version numbers of ops and ops-scenario, in
-   the form `ops <major>.<minor>.<patch> & ops-scenario <major>.<minor>.<patch>`
-   and a brief summary of the main changes in the releases. If the release only
-   includes one of `ops` and `ops-scenario`, leave out the other one.
+To make a release of the `ops` and/or `ops-scenario` packages, do the following:
+
+1. Check if there's a `chore: update charm pins` auto-generated PR in the queue.
+   If it looks good, merge it and check that tests still pass. If needed, you
+   can re-trigger the `Update Charm Pins` workflow manually to ensure latest
+   charms and ops get tested.
+2. Visit the [releases page on GitHub](https://github.com/canonical/operator/releases).
+3. Click "Draft a new release"
+4. The "Release Title" is the full version numbers of ops and/or ops-scenario,
+   in the form `ops <major>.<minor>.<patch> and ops-scenario <major>.<minor>.<patch>`
+   and a brief summary of the main changes in the release.
    For example: `ops 2.3.12 Bug fixes for the Juju foobar feature when using Python 3.12`
-4. Have the release create a new tag, in the form `<major>.<minor>.<patch>`
-5. Use the "Generate Release Notes" button to get a copy of the changes into the
-   notes field.
-6. Group the changes first by package (`ops` and `ops-scenario`) and then by
-   the commit type (feat, fix, etc.) and use full names (e.g., "Features",
-   not "feat") for group headings. Strip the commit type prefix from the bullet point,
-   and capitalise the first word. Strip the username (who did each commit) if the
-   author is a member of the Charm Tech team.
-   For example: the PR `docs: clarify where StoredState is stored`
-   becomes `* Clarify where StoredState is stored` in the "Documentation" section.
-7. Where appropriate, collapse multiple tightly related bullet points into a
-   single point that refers to multiple commits.
-8. Create a new branch, and copy this text to the [CHANGES.md](CHANGES.md) file,
-   stripping out links, who did each commit, the new contributor list, and the
-   link to the full changelog.
-9. Change [version.py](ops/version.py)'s `version` to the
-   [appropriate string](https://semver.org/).
-10. Check if there's a `chore: update charm pins` auto-generated PR in the queue. If it looks
-   good, merge it and check that tests still pass. If needed, you can re-trigger the
-   `Update Charm Pins` workflow manually to ensure latest charms and ops get tested.
-11. Add, commit, and push, and open a PR to get the changelog and version bump
+5. Have the release create a new tag, in the form `<major>.<minor>.<patch>` for
+   `ops` and `scenario-<major>.<minor>.<patch>` for `ops-scenario`. If releasing
+   both packages, use the ops tag.
+6. Use the "Generate Release Notes" button to get a copy of the changes into the
+   notes field. The 'Release Documentation' section below details the form that
+   the release notes and changelog should take.
```

Comment:
Worth adding a note about choosing the previous release tag to compare to?

I think the rule is: find the previous release tag for each thing you're releasing, and if there is more than one (i.e. an ops only release and a scenario only release), use the older of the two.

Say you had an ops release, then a scenario release, and are now doing a joint release -- you'd want to choose the ops release to generate release notes.

But you'd need to filter out any scenario PRs from before the last scenario release, since they'd have been covered already.

Reply:
We can add it to the release instructions! "If you're the first one to do a patch release, update the release instructions with what you did and remove this directive."

Reply:
Clarification: mostly joking. I'll hit approve now as I don't think any of this is worth blocking merging over (though you have two approvals anyway)